### ISSUE TITLE
How to restore the database if I have the salt key and previous database folders 

### ISSUE THREAD

**Issue by Carolhug** [nilsteampassnet/TeamPass#2024]
### Steps to reproduce
1. Teampass database is crashed because we are doing the docker cluster maintenance. 
2. Teampass database is not able to connect 
3. Re-installed the Teampass database and teampass 

### Expected behaviour
copied  the previous /var/lib/mysql/teampass file to the new mysql database instance /var/lib/mysql/teampass 
restart the new mysql database
put the previous salt key into /teampass/sk 

### Actual behaviour
now if I access to my teampass website, the page becomes to blank, and from the webpage network analyzed, it is 500 internal server error. 
### Server configuration
**Docker version 1.12.6, build 0fdc778/1.12.6,[root@ooc-master01 sk]# kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1+5115d708d7", GitCommit:"fff65cf", GitTreeState:"clean", BuildDate:"2017-11-03T15:44:55Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1+5115d708d7", GitCommit:"fff65cf", GitTreeState:"clean", BuildDate:"2017-11-03T15:44:55Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
 **:

**:# apachectl -v                                                                                                                                                                                                     
Server version: Apache/2.4.7 (Ubuntu)                                                                                                                                                                                                     
Server built:   Jul 15 2016 15:34:04   **

**mysql 5.7:**

**PHP version:**

**Teampass version:** 

**Teampass configuration file:**

**fresh install:**
PLEASE attach to this message the file `/includes/config/tp.config.php`.

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
```
Insert your webserver log here
```

#### Log from the web-browser developer console (CTRL + SHIFT + i)
```
Insert the log here and especially the answer of the query that failed.
```

**Comment by nilsteampassnet** [comment]
You also need file `csrfp.config.php'

Please check http://teampass.readthedocs.io/en/latest/errors/#users-and-folders-lists-are-empty that may help solving your issue.

**Comment by Carolhug** [comment]
Hi Nils,


Thank you! I'm able to open the page now, I fixed by change the database server address. Why I can't open the page, it is because I copied whole the previous config file to my new teampass Apache server, and forgot the modify the database server name since it is the new one which I deploy in my container.


But here is a new problem, in the database, the user table is empty, but other data are there, could you please advise me how to recover all my user data? The two files of temapass_users.frm and teampass_users.ibd are copied from previous database server, the file size still the same, please check below output:

[root@ooc-master01 teampass]# ls -la ../../data/teampass/teampass_users.ibd
-rw-r----- 1 33 33 114688 Oct 17 18:33 ../../data/teampass/teampass_users.ibd
[root@ooc-master01 teampass]# ls -la ../../data/teampass/teampass_users.frm
-rw-r----- 1 33 33 18278 Sep 28 16:18 ../../data/teampass/teampass_users.frm    --------previous teampass iser database file
[root@ooc-master01 teampass]# ls -la teampass_users.*
-rw-r----- 1 27 root  18278 Nov 20 17:17 teampass_users.frm
-rw-r----- 1 27 root 114688 Nov 21 10:02 teampass_users.ibd  --------------Current teampass user database file


Thanks

- Carol

________________________________

**Comment by Carolhug** [comment]
Oh, and if I trying to create a new user, I'll get the error:


[Tue Nov 21 10:22:28.301516 2017] [:error] [pid 26] [client 10.131.0.1:57634] PHP Notice:  Undefined variable: link in /teampass/www/includes/libraries/Database/Meekrodb/db.class.php on line 618, referer: https://teampass.paas.anim.odw.com.cn/index.php?page=manage_users
[Tue Nov 21 10:22:28.301576 2017] [:error] [pid 26] [client 10.131.0.1:57634] PHP Warning:  mysqli_real_escape_string() expects parameter 1 to be mysqli, null given in /teampass/www/includes/libraries/Database/Meekrodb/db.class.php on line 618, referer: https://teampass.paas.com.cn/index.php?page=manage_users



- Carol

________________________________

**Comment by Carolhug** [comment]
Okay, I found a solution, put the database to a new mysql server(not in the docker), then restored the database, replaced the /var/lib/mysql/data teampass folder and ibdata1/ib_buffer_pool/ibtmp1 file, restart the mysql server, then it's able to query all the data which I stored. 
After that, use mysqldump command export the data. Installed a whole new teampass and mysql in kubernet server, import the database from .sql file and copied back the salt key and put it under /teampass/sk/ folder. 

Now everything is working well

Thanks for your help here :+1:
